core/src/main/java/org/testcontainers/containers/image/ImageData.java

```diff
@@ -0,0 +1,34 @@
+package org.testcontainers.containers.image;
+
+import com.github.dockerjava.api.command.InspectImageResponse;
+import com.github.dockerjava.api.model.Image;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+@Data
+@RequiredArgsConstructor(access = AccessLevel.MODULE)
+public class ImageData {
```

Comment:
I would suggest to make it an interface, with `DockerJavaImageData` implementation having `public InspectImageResponse getInspectImageResponse` in case it gets more field we didn't support and the users will want to cast and get the original response

Reply:
I'm not sure this is a good idea - ImageData can be instantiated from either Image (most common case) or InspectImageResponse (rare scenario), and it's not possible to determine at runtime how it was created.
If the image is present in the local Docker cache, ImageData will be instantiated from an Image instance. If the image is not present in the cache, it will be downloaded and ImageData will be instantiated from InspectImageResponse.
ImagePullPolicy can't assume anything about the source of the data, because it can even change between two containers being instantiated in the same test class.

Furthermore, ImagePullPolicy is only ever evaluated for images that were instantiated from Image, never for InspectImageResponse.

Reply:
Sorry ignore the last statement. ImagePullPolicy is evaluated for both types of images, but cannot assume anything about how the ImageData representation was created.

Reply:
I would still suggest to replace it with an interface, with two implementations. 
Class here makes it impossible to access the underlying object (be it `Image` or `InspectImageResponse`) and will force us to change & release the code in case someone will need to access some rare field of the mentioned docker-java's classes.

Reply:
I've changed the implementation such that it no longer iterates all existing images but instead inspects only the requested images. That way I always have the original InspectImageResponse that I can provide back to the user, if needed.